conn.go

```diff
@@ -68,6 +68,7 @@ type Conn struct {
 	writeBuf       []byte
 	writeHeaderBuf [8]byte
 	writeHeader    header
+	closeSent      bool
```

Comment:
Review: I moved this here so that the it's clearer that `writeFrameMu` is what's protecting it.